### PR TITLE
Templates for documentation

### PR DESCRIPTION
**Pull request by cosunae**
Template for building a documentation with sphinx. Inspire in xarray. 
* Change the default (DARGLINT) convention to numpy style guide. 
* Use the sphinx_book_theme by the https://executablebooks.org/en/latest/ (same as xarray)

It can be built with 
```
cd doc
make html
```

### CODE REVIEW

**Comment by cosunae** [comment]
No preferences on the style guide nor the theme. I just picked it from xarray, so please comment if you have a better choice

**Comment by victoria-cherkas** [comment]
```
(idp) bash-4.2$ make html
Running Sphinx v5.0.0
making output directory... done
WARNING: html_static_path entry '_static' does not exist

Theme error:
no theme named 'sphinx_book_theme' found (missing theme.conf?)
```

Am I missing something here? I am in doc branch and doc directory

**Comment by victoria-cherkas** [comment]
Ah I simply had to pip install sphinx_book_theme.

File: doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+=============
+
+This page provides an auto-generated summary of iconarray's API.
+
+.. autosummary::
+   :toctree: generated/
+
+   core
+   core.utilities 
+   core.utilities.ind_from_latlon
```
Comment (victoria-cherkas) [inline doc/index.rst:23]: When i ran make html, I didnt see the documentation until i changed these 3 lines to 

   iconarray.core
   iconarray.core.utilities 
   iconarray.core.utilities.ind_from_latlon

Comment (cosunae) [reply doc/index.rst]: thanks, I fixed it

**Comment by cosunae** [comment]
sounds good to me then, we can keep it at the github action. 
Shall we merge or wait until #8 is ready?

**Comment by victoria-cherkas** [comment]
I would like to wait until https://github.com/C2SM/iconarray/pull/8 is ready, which i hope is very soon...
The problem I have is the environment -  when I build the documentation locally in my conda env (build with the environment.yml) then it builds the documenation correctly, but outside of this conda environment (just pip installing sphinx and the theme etc) then it builds the documentation with the pages and all, but minus the content (the docstrings converted to documentation). And this is also what happens in the github action. Not sure which additional packages are required, or if I should just setup the github action to install the conda environment before make build.

**Pull request merged**